### PR TITLE
fix(comp: divider): dynamic slot

### LINKED ISSUE #76: [comp: Divider] dynamic slots auto rerender
### Any additional comments? (optional)

---
| Environment | Info |
|---|---|
| @idux |     |
| Browser |     |

### PR DESCRIPTION
**Pull request by LaamGinghong**
fix #76

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IduxFE/components/blob/main/docs/contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

File: packages/components/divider/src/Divider.vue [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
       <slot />
     </span>
   </div>
 </template>
 <script lang="ts">
-import { computed, defineComponent } from 'vue'
-import { PropTypes } from '@idux/cdk/utils'
-import { useGlobalConfig } from '@idux/components/core/config'
-
+import type { Ref } from 'vue'
```
Comment (danranVm) [inline packages/components/divider/src/Divider.vue:9]:
```suggestion
import type { Ref, SetupContext  } from 'vue'
```

File: packages/components/divider/src/Divider.vue [hunk 2]
```diff
@@ -1,18 +1,20 @@
 <template>
   <div class="ix-divider" :class="className">
-    <span v-if="$slots.default" class="ix-divider-inner-text">
+    <span v-if="$slots.default && type !== 'vertical'" class="ix-divider-inner-text">
```
Comment (danranVm) [inline packages/components/divider/src/Divider.vue:3]: 这个判断没必要加？用户真传了就自己负责。

File: packages/components/core/utils/vNode.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  return node
+}
+
+/**
+ * determine if the element is a valid element type rather than fragments and comment e.g. <template> v-if
+ * @param node {VNode} node to be tested
+ */
+export const isValidElementNode = (node: VNodeChild) => !(isFragment(node) || isComment(node))
+
+export const getFirstValidNode = (nodes: VNodeChild | null, maxDepth = 3): ReturnType<typeof getChildren> => {
```
Comment (danranVm) [inline packages/components/core/utils/vNode.ts:35]: 这个函数加个注释，尤其是 maxDepth 要做下说名，指出默认值。

**Review by danranVm — CHANGES_REQUESTED** [review]

**Comment by LaamGinghong** [comment]
@danranVm 好了

**Review by danranVm — APPROVED** [review]

**Comment by danranVm** [comment]
PGTM.

**Pull request merged**